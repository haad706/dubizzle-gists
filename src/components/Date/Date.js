import styled from "styled-components";
import PropTypes from "prop-types";
import { getFormattedDate } from "utils/helper";

export const Date = ({ label, time }) => {
  return (
    <Wrapper>
      <p>{label}:</p>
      <p>{getFormattedDate(time)}</p>
    </Wrapper>
  );
};
const Wrapper = styled.div`
  display: inline-flex;
  gap: 0.5rem;
  justify-content: center;
  align-items: center;
  font-weight: 400;
  font-size: 13px;
`;

Date.propTypes = {
  label: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
};

export default Date;
