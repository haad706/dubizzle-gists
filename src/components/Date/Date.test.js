import { render } from "@testing-library/react";
import Date from "./Date";

describe("Date", () => {
  it("should render successfully", () => {
    const { baseElement } = render(
      <Date time="2023-05-24T06:51:40Z" label="Created At" />
    );
    expect(baseElement).toBeTruthy();
  });
  it("should match snapshot", () => {
    const { asFragment } = render(
      <Date time="2023-05-24T06:51:40Z" label="Created At" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
