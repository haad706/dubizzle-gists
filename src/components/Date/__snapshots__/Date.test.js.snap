// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bgqQcB fppZgW"
  >
    <p>
      Created At:
    </p>
    <p>
      24/05/2023
    </p>
  </div>
</DocumentFragment>
`;
